types/akamai-edgeworkers/index.d.ts

```diff
@@ -1359,3 +1359,125 @@ declare module "crypto" {
 
     export { crypto };
 }
+
+/**
+ * HtmlRewriter rewrites HTML documents by parsing and constructing the DOM.
+ * It allows for registering callbacks on CSS selectors that execute when
+ * the parser encounters an element matching the selector, enabling modification
+ * of tag attributes, insertion of new content around the element, or removal of the element.
+ */
+declare module "html-rewriter" {
+    import { ReadableStream, WritableStream } from "streams";
+
+    class HtmlRewritingStream implements GenericHtmlRewritingStream {
+        readonly writable: WritableStream;
+        readonly readable: ReadableStream;
+
+        /**
+         * Constructor for a new HtmlRewritingStream object
+         */
+        constructor();
+
+        /**
+         * Add one or more handlers using onElement(). The handlers call functions on their argument to modify the stream.
+         * @param selector is a string CSS selector that specifies when the handler should run.
+         * @param handler is a function that runs when the selector matches.
+         * When the HtmlRewritingStream calls the handler, it passes an Element object as an argument.
+         */
+        onElement(selector: string, handler: (element: Element) => void): void;
+    }
+
+    interface GenericHtmlRewritingStream {
+        readonly readable: ReadableStream;
+        readonly writable: WritableStream;
+    }
+
+    /**
+     * The Element object is an argument to the handler registered with onElement(),
+     * the handler calls functions on the Element to modify the output stream.
+     */
+    interface Element {
+        /**
+         * Insert new content immediately after the end tag of the matched element.
+         * @param tag is the new text to insert.
+         * @param trailing_opt When the value is true, elements that are missing a close tag will have one inserted.
```

Comment:
`trailing_opt` is not boolean, so `When the value is true` is a bit misleading. Perhaps something type-nonspecific would be more accurate: _Controls whether elements missing a close tag should have one inserted_.